ccl/changefeedccl: TestChangefeedSchemaChangeBackfillCheckpoint failed

ccl/changefeedccl.TestChangefeedSchemaChangeBackfillCheckpoint [failed](https://tanzanite.cluster.engflow.com/invocations/default/fcf3cfe4-e213-4b7c-9ace-6a1835bb9981?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [1d2534b831775dd84edcf7372d38e35ed4903278](https://github.com/cockroachdb/cockroach/commits/1d2534b831775dd84edcf7372d38e35ed4903278):


```
=== RUN   TestChangefeedSchemaChangeBackfillCheckpoint
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedSchemaChangeBackfillCheckpoint2647998574
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:2167: random seed: 8532591158834740398
    changefeed_test.go:2433: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaChangeBackfillCheckpoint2647998574
--- FAIL: TestChangefeedSchemaChangeBackfillCheckpoint (52.31s)
=== RUN   TestChangefeedSchemaChangeBackfillCheckpoint/webhook
    helpers_test.go:867: making server as system tenant
    helpers_test.go:942: making webhook feed factory
    changefeed_test.go:2317: condition failed to evaluate within 45s: from changefeed_test.go:2321: waiting for checkpoint
    --- FAIL: TestChangefeedSchemaChangeBackfillCheckpoint/webhook (52.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaChangeBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35742

--- a/pkg/ccl/changefeedccl/parallel_io.go
+++ b/pkg/ccl/changefeedccl/parallel_io.go
@@ -15,6 +15,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/intsets"
 	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
@@ -84,15 +85,18 @@ func NewParallelIO(
 	metrics metricsRecorder,
 	settings *cluster.Settings,
 ) *ParallelIO {
+	quota := uint64(requestQuota.Get(&settings.SV))
 	wg := ctxgroup.WithContext(ctx)
 	io := &ParallelIO{
 		retryOpts: retryOpts,
 		wg:        wg,
 		metrics:   metrics,
 		ioHandler: handler,
-		quota:     quotapool.NewIntPool("changefeed-parallel-io", uint64(requestQuota.Get(&settings.SV))),
-		requestCh: make(chan AdmittedIORequest, numWorkers),
-		resultCh:  make(chan IOResult, numWorkers),
+		quota:     quotapool.NewIntPool("changefeed-parallel-io", quota),
+		// NB: The size of these channels should not be less than the quota. This prevents the producer from
+		// blocking on sending requests which have been admitted.
+		requestCh: make(chan AdmittedIORequest, quota),
+		resultCh:  make(chan IOResult, quota),
 		doneCh:    make(chan struct{}),
 	}
 
@@ -161,8 +165,10 @@ var requestQuota = settings.RegisterIntSetting(
 	"changefeed.parallel_io.request_quota",
 	"the number of requests which can be admitted into the parallelio"+
 		" system before blocking the producer",
-	128,
-	settings.PositiveInt,
+	int64(util.ConstantWithMetamorphicTestChoice(
+		"changefeed.parallel_io.request_quota",
+		128, 16, 32, 64, 256).(int)),
+	settings.IntInRange(1, 256),
 	settings.WithVisibility(settings.Reserved),
 )
 

